dapp/src/components/Header/ConnectWallet.tsx

```diff
@@ -63,7 +70,47 @@ export default function ConnectWallet() {
     )
   }
 
-  return (
+  return isMobile ? (
+    <Menu>
+      {({ isOpen }) => (
+        <>
+          <MenuButton
+            as={Button}
+            variant="card"
+            leftIcon={<Icon as={BitcoinIcon} boxSize={6} color="brand.400" />}
+            rightIcon={isOpen ? <IconChevronUp /> : <IconChevronDown />}
+          >
+            <TextMd color="brand.400">{truncateAddress(address)}</TextMd>
+          </MenuButton>
+          <MenuList bg="gold.200">
+            <MenuItem
+              {...styles}
+              icon={<Icon as={IconCopy} boxSize={5} />}
+              onClick={onCopy}
+            >
+              {hasCopied ? "Address copied" : "Copy Address"}
+            </MenuItem>
+            {isChangeAccountFeatureSupported(embeddedApp) && (
+              <MenuItem
+                {...styles}
+                icon={<Icon as={IconUserCode} boxSize={5} />}
+                onClick={() => handleConnectWallet(true)}
+              >
+                Change account
+              </MenuItem>
+            )}
+            <MenuItem
+              {...styles}
+              icon={<Icon as={IconLogout} boxSize={5} />}
+              onClick={onDisconnect}
+            >
+              Disconnect
+            </MenuItem>
```

Comment:
We duplicate the components here. So maybe we can declare the menu items in an array. What do you think?

```
 const MENU_OPTIONS = [
  {
    icon: IconCopy,
    label: "Copy Address",
    onClick: onCopy,
  },
  ...(isChangeAccountFeatureSupported(embeddedApp)
    ? [
        {
          icon: IconUserCode,
          label: "Change account",
          onClick: () => handleConnectWallet(true),
        },
      ]
    : []),
  {
    icon: IconLogout,
    label: "Disconnect",
    onClick: onDisconnect,
  },
]
```

```
{MENU_OPTIONS.map(({ icon, label, onClick }) => (
  <MenuItem
    {...styles}
    icon={<Icon as={icon} boxSize={5} />}
    onClick={onClick}
  >
    {label}
  </MenuItem>
))}
```

Reply:
68c5c99